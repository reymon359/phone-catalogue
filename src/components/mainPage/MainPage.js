import React, { Component } from 'react';

import CardList from '../cardList/CardList';
import CardDetails from '../cardDetails/CardDetails'
import Header from '../header/Header';
import Intro from '../intro/Intro'
import { BrowserRouter, Route,  Switch, Link} from 'react-router-dom';
export class MainPage extends Component {
    componentDidMount() {
        this.props.onRequestPhones();
    }

    render() {
        const { isPending , phones } = this.props;

        return (
            <div className="tc white">
            <BrowserRouter>
                <Link className='link dim' to='/' > <Header/></Link>
                <Intro />
               <Switch>
               <Route exact path="/" render={() => (
                    isPending ? <h1>Loading...</h1> :
                    <CardList phones={phones} />
               )} />
                <Route path="/details/:id" render={(props) => (
                    <CardDetails id={props.match.params.phoneId} />
                )} />
                </Switch>
            </BrowserRouter>
            </div>
        );
    }
}

export default MainPage;
