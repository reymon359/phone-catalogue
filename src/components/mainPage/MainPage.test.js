import React from 'react';
import { shallow } from 'enzyme';
import MainPage from './MainPage';

let wrapper;
beforeEach(() => {
    const mockProps = {
        onRequestPhones: jest.fn(),
        phones: [],
        searchField: '',
        isPending: false
    };
    wrapper = shallow(<MainPage {...mockProps} />);
});
it('should render MainPage without crashing', () => {
    expect(wrapper).toMatchSnapshot();
});
