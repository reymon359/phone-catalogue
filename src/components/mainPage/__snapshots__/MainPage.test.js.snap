// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MainPage without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainPage
    isPending={false}
    onRequestPhones={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    phones={Array []}
    searchField=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <BrowserRouter>
        <ForwardRef
          className="link dim"
          to="/"
        >
           
          <Header />
        </ForwardRef>
        <Intro />
        <Switch>
          <Route
            exact={true}
            path="/"
            render={[Function]}
          />
          <Route
            path="/details/:id"
            render={[Function]}
          />
        </Switch>
      </BrowserRouter>,
      "className": "tc white",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ForwardRef
            className="link dim"
            to="/"
          >
             
            <Header />
          </ForwardRef>,
          <Intro />,
          <Switch>
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />
            <Route
              path="/details/:id"
              render={[Function]}
            />
          </Switch>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              " ",
              <Header />,
            ],
            "className": "link dim",
            "to": "/",
          },
          "ref": null,
          "rendered": Array [
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "Link",
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/"
                render={[Function]}
              />,
              <Route
                path="/details/:id"
                render={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/details/:id",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <BrowserRouter>
          <ForwardRef
            className="link dim"
            to="/"
          >
             
            <Header />
          </ForwardRef>
          <Intro />
          <Switch>
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />
            <Route
              path="/details/:id"
              render={[Function]}
            />
          </Switch>
        </BrowserRouter>,
        "className": "tc white",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <ForwardRef
              className="link dim"
              to="/"
            >
               
              <Header />
            </ForwardRef>,
            <Intro />,
            <Switch>
              <Route
                exact={true}
                path="/"
                render={[Function]}
              />
              <Route
                path="/details/:id"
                render={[Function]}
              />
            </Switch>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                " ",
                <Header />,
              ],
              "className": "link dim",
              "to": "/",
            },
            "ref": null,
            "rendered": Array [
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "Link",
              "propTypes": Object {
                "innerRef": [Function],
                "onClick": [Function],
                "replace": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  exact={true}
                  path="/"
                  render={[Function]}
                />,
                <Route
                  path="/details/:id"
                  render={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "exact": true,
                  "path": "/",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "path": "/details/:id",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
